cypress/support/pages/roles.ts

```diff
@@ -0,0 +1,47 @@
+import { keys } from "cypress/types/lodash";
+import { verify } from "cypress/types/sinon";
+import { randomNumber, role, smallRandomNumber, team, user } from "../../constants/globalConstants"
+
+//general data
+
+const name_1 = 'EA';
+const role_name = name_1+randomNumber;
+
+
+
+ //create a role
+Cypress.Commands.add('createARole', () => {
+  
+   //we are creating a super admin role
+
+   cy.get('#roles-link').click();
+   cy.get('#root > section > main > div > div > div > div > div.ant-table-title > header > div.cta-btns > a').click();
+   const name = 'EA' +randomNumber
+   cy.get('#name').type(role_name);
+   cy.get('#displayName').type(role_name);
+   cy.get('.ant-select-selection-item').type('{downarrow}');
+   cy.xpath('/html/body/div[3]/div/div/div/div[2]/div[1]/div/div/div[1]/div').click();
+   cy.get('#wipLimit').type(''+smallRandomNumber);
+   cy.get('#description').type('test');
+   cy.get('#isSupervisor > label:nth-child(1) > span.ant-radio > input').click();
+   cy.get('#isClinical > :nth-child(1)').click();
+   cy.get('#isRoutable > :nth-child(1) > :nth-child(2)').click();
+   cy.get('#submit-create-btn').click().wait(2000);
+   cy.get('#permissions-link').click();
+   cy.get('[type="checkbox"]').click({ multiple: true, force: true });
+   cy.wait(2000);
```

Comment:
cypress has an extremely robust system for waiting for events to complete before moving forward executing code. We should try as much as possible to make our waits implicit instead of explicit.

https://docs.cypress.io/api/commands/wait#Alias

check out their example for waiting on a route alias - that is, waiting for a request to finish directly